pkg/notary/registry.go

```diff
@@ -101,29 +78,25 @@ func isDigestReference(reference string) bool {
 	return index != -1
 }
 
-func getRemoteOpts(authenticator authn.Authenticator) ([]gcrremote.Option, error) {
-	remoteOpts := []gcrremote.Option{}
-	remoteOpts = append(remoteOpts, gcrremote.WithAuth(authenticator))
+func resolveDigestCrane(repo notationregistry.Repository, remoteOpts []gcrremote.Option, ref name.Reference) error {
+	_, err := repo.Resolve(context.Background(), ref.Name())
+	if err != nil {
+		return err
+	}
+	return nil
+}
 
+func gcrRemoteOpts(remoteOpts []gcrremote.Option) []gcrremote.Option {
 	pusher, err := gcrremote.NewPusher(remoteOpts...)
 	if err != nil {
-		return nil, err
+		return nil
 	}
 	remoteOpts = append(remoteOpts, gcrremote.Reuse(pusher))
 
 	puller, err := gcrremote.NewPuller(remoteOpts...)
 	if err != nil {
-		return nil, err
+		return nil
 	}
```

Comment:
Add this to the client package as well.

What you should do is, add 1 more function to the interface (for a temporary basic alongside the original, something like

```golang
// BuildCosignRemoteOption builds remote.Option based on client.
	BuildCosignRemoteOption(context.Context) remote.Option

	// BuildGCRRemoteOption builds remote.Option based on client.
	BuildGCRRemoteOption(context.Context) ([]gcrremote.Option, error)
```

We will update this to use the same thing once we can upgrade to Cosign 2.1.1